internal/generator/vector/output/loki/loki.go

```diff
@@ -233,3 +235,40 @@ func BasicAuth(o logging.OutputSpec, secret *corev1.Secret) []Element {
 
 	return conf
 }
+
+func BearerTokenAuth(o logging.OutputSpec, secret *corev1.Secret) []Element {
+	conf := []Element{}
+	if o.Secret != nil {
+		// external loki, use user provided secret
+		// the secret contains the path of the token
+		if security.HasBearerTokenFileKey(secret) {
+			conf = append(conf, BasicAuthConf{
+				Desc:        "Bearer Auth Config",
+				ComponentID: strings.ToLower(vectorhelpers.Replacer.Replace(o.Name)),
+			})
+			token, _ := ioutil.ReadFile(security.GetFromSecret(secret, constants.BearerTokenFileKey))
```

Comment:
it'd be better to handle the error here

Reply:
will add this to the verify step before conf generation

Reply:
handled